lib/reek/ast/object_refs.rb

```diff
@@ -1,33 +1,30 @@
 module Reek
+  # @api private
   module AST
+    ObjectRef = Struct.new(:name)
     #
     # Manages and counts the references out of a method to other objects.
     #
-    # @api private
-    class ObjectRefs  # :nodoc:
+    class ObjectRefs
       def initialize
-        @refs = Hash.new(0)
+        @refs = Hash.new { |refs, name| refs[name] = [] }
       end
 
-      def record_reference_to(exp)
-        @refs[exp] += 1
+      def most_popular
+        max = @refs.values.map(&:size).max
+        @refs.select { |_name, refs| refs.size == max }
       end
 
-      def references_to(exp)
-        @refs[exp]
+      def record_reference_to(name)
+        @refs[name] << ObjectRef.new(name)
```

Comment:
The counters is what we had previously – but for #480 we need to track line numbers, so the fix for that will have `ObjectRef`s that have a `:name` and a `:line` – and we need FeatureEnvy to get the distinct line numbers – see e0b75b7 + 769ae0b (will be rebased, of course).